cmd/juju/commands/debughooks.go

```diff
@@ -174,36 +185,23 @@ func (c *debugHooksCommand) validateHooksOrActions() error {
 	return nil
 }
 
-func (c *debugHooksCommand) getValidActions(appName string) (set.Strings, error) {
-	allActions, err := c.actionsAPI.ApplicationCharmActions(appName)
-	if err != nil {
-		return nil, err
-	}
-
+func (c *debugHooksCommand) getValidActions(ch charm.Charm) (set.Strings, error) {
 	validActions := set.NewStrings()
-	for name := range allActions {
+	for name := range ch.Actions().ActionSpecs {
 		validActions.Add(name)
 	}
 	return validActions, nil
 }
 
-func (c *debugHooksCommand) getValidHooks(appName string) (set.Strings, error) {
-	relations, err := c.charmRelationsAPI.CharmRelations(appName)
-	if err != nil {
-		return nil, err
-	}
-
-	validHooks := set.NewStrings()
-	for _, hook := range hooks.UnitHooks() {
-		validHooks.Add(string(hook))
-	}
-	for _, relation := range relations {
-		for _, hook := range hooks.RelationHooks() {
-			hook := fmt.Sprintf("%s-%s", relation, hook)
-			validHooks.Add(hook)
-		}
-	}
-	return validHooks, nil
+func (c *debugHooksCommand) getValidHooks(ch charm.Charm) (set.Strings, error) {
+	hooks := set.Strings(ch.Meta().Hooks())
+	return hooks.Union(set.NewStrings(
+		"juju-info-relation-broken",
+		"juju-info-relation-changed",
+		"juju-info-relation-created",
+		"juju-info-relation-departed",
+		"juju-info-relation-joined",
+	)), nil
```

Comment:
We should still construct the "juju-info" hooks by iterating over hooks.RelationHooks()

```
for _, hook := range hooks.RelationHooks() {
	hook := fmt.Sprintf("juju-info-%s", relation, hook)
	hooks.Add(hook)
}
```